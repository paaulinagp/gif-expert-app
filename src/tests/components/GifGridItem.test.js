import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem  from '../../components/GifGridItem';

describe('Pruebas del componente <GifGridItem /> ', () => {
  const title = 'Un titulo';
  const url = 'http://localhost/img.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('debe de mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de tener un parrafo con el titulo', () => {
    const p = wrapper.find('p');
    expect( p.text().trim() ).toBe(title);
  });

  test('debe de tener una imagen igual al url y al alt de los props', () => {
    const img = wrapper.find('img');
    expect( img.props().src ).toBe(url);
    expect( img.props().alt ).toBe(title);
  });

  test('debe de tener animate__fadeIn', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');

    expect(className.includes('animate__fadeIn')).toBe(true);
  });
});
