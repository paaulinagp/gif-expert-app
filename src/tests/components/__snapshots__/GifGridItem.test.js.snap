// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifGridItem />  debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__delay-3s"
>
  <img
    alt="Un titulo"
    src="http://localhost/img.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
